packages/loader/container-loader/src/memoryBlobStorage.ts

```diff
@@ -0,0 +1,77 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { ICreateBlobResponse } from "@fluidframework/protocol-definitions";
+import { bufferToString, stringToBuffer } from "@fluid-internal/client-utils";
+import { assert, isObject } from "@fluidframework/core-utils/internal";
+// eslint-disable-next-line import/no-deprecated
+import type { IDetachedBlobStorage } from "./loader.js";
+
+const MemoryDetachedBlobStorageIdentifier = Symbol();
+
+// eslint-disable-next-line import/no-deprecated
+interface MemoryDetachedBlobStorage extends IDetachedBlobStorage {
+	[MemoryDetachedBlobStorageIdentifier]: typeof MemoryDetachedBlobStorageIdentifier;
+	initialize(attachmentBlobs: string[]): void;
+	serialize(): string | undefined;
+}
+
+function isMemoryDetachedBlobStorage(
+	// eslint-disable-next-line import/no-deprecated
+	detachedStorage: IDetachedBlobStorage,
+): detachedStorage is MemoryDetachedBlobStorage {
+	return (
+		isObject(detachedStorage) &&
+		MemoryDetachedBlobStorageIdentifier in detachedStorage &&
+		detachedStorage[MemoryDetachedBlobStorageIdentifier] === MemoryDetachedBlobStorageIdentifier
+	);
+}
+
+export function serializeMemoryDetachedBlobStorage(
+	// eslint-disable-next-line import/no-deprecated
+	detachedStorage: IDetachedBlobStorage,
+): string | undefined {
+	if (detachedStorage.size > 0 && isMemoryDetachedBlobStorage(detachedStorage)) {
+		return detachedStorage.serialize();
+	}
+}
+
+export function tryInitializeMemoryDetachedBlobStorage(
+	// eslint-disable-next-line import/no-deprecated
+	detachedStorage: IDetachedBlobStorage,
+	attachmentBlobs: string,
+) {
+	if (!isMemoryDetachedBlobStorage(detachedStorage)) {
+		throw new Error("Blob storage not memory blob storage and cannot be initialize.");
+	}
+
+	assert(detachedStorage.size === 0, "Blob storage already initialized");
+	const maybeAttachmentBlobs = JSON.parse(attachmentBlobs);
+	assert(Array.isArray(maybeAttachmentBlobs), "Invalid attachmentBlobs");
+
+	detachedStorage.initialize(maybeAttachmentBlobs);
+}
+
+// eslint-disable-next-line import/no-deprecated
+export function createMemoryDetachedBlobStorage(): IDetachedBlobStorage {
+	const blobs: ArrayBufferLike[] = [];
+	const storage: MemoryDetachedBlobStorage = {
+		[MemoryDetachedBlobStorageIdentifier]: MemoryDetachedBlobStorageIdentifier,
+		createBlob: async (file: ArrayBufferLike): Promise<ICreateBlobResponse> => ({
+			id: `${blobs.push(file) - 1}`,
+		}),
+		readBlob: async (id: string): Promise<ArrayBufferLike> => blobs[Number(id)],
```

Comment:
Did a little code inspection, but want to confirm--does FF control usage of this function? If not, seems like we should provide better errors here for incorrect usage. I only found references in `attachment.ts` though, so I think fine as-is.